core/drivers/crypto/versal/authenc.c

```diff
@@ -400,9 +406,11 @@ static TEE_Result do_init(struct drvcrypt_authenc_init *dinit)
 
 	return TEE_SUCCESS;
 error:
-	free(key.buf);
-	free(init_buf.buf);
-	free(nonce.buf);
+	versal_mbox_free(&nonce);
```

Comment:
since `versal_mbox_free()` zeroes the internal buffer reference, you don't need these numerous branch label. The below instructions are enough:
```c
error:
	versal_mbox_free(&nonce);
	versal_mbox_free(&init_buf);
	versal_mbox_free(&key);

	return ret;
```

Reply:
Fixed by fixup commit 446148d